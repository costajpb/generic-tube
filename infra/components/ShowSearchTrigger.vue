<script setup lang="ts">
    defineProps<{
        anchor: string
    }>()
</script>

<template>
    <a :class="classes.trigger" :href="'#' + anchor">Search</a>
</template>

<style lang="postcss" module="classes">
    .trigger {
        text-indent: 100vw;
        overflow: hidden;
        position: relative;
        height: var(--size-7);
        width: var(--size-7);
        display: inline-block;
        background: var(--color-background-default);
        border-radius: var(--radius-round);
        vertical-align: middle;

        &::before {
            content: "üîç";
            font-size: var(--font-size-2);
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
            text-indent: 0;
        }

        @media (min-width: 700px) {
            display: none
        }
    }
</style>