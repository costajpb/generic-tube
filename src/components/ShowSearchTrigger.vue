<script setup lang="ts">
    defineProps<{
        anchor: string
    }>()
</script>

<template>
    <a :href="'#' + anchor">Search</a>
</template>

<style lang="postcss" scoped>
    a {
        font-size: 0;
        position: relative;
        height: var(--size-3);
        width: var(--size-3);
        /* FIXME */
        background: white;
        border-radius: var(--radius-round);
    }

    a::before {
        content: "üîç";
        font-size: var(--font-size-2);
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
    }
</style>