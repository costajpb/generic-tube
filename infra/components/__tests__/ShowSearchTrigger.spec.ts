import snapshowWrapper from "@/infra/__tests__/snapshotWrapper"
import ShowSearchTrigger from "../ShowSearchTrigger.vue"

describe('infra/components/ShowSearchTrigger', () => {
    it('should render', () => {
        expect(snapshowWrapper(ShowSearchTrigger)).toMatchSnapshot()
    })
})