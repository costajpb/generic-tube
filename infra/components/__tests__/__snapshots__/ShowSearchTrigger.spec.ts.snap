// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`infra/components/ShowSearchTrigger > should render 1`] = `
<a
  class="_trigger_22e8e9"
  href="#undefined"
>
  Search
</a>
`;
